services/web/src/modals/LoginAsUser.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { request } from 'utils/api';
+
+import { Message, Modal, Button } from 'semantic';
+import modal from 'helpers/modal';
+
+@modal
+export default class LoginAsUser extends React.Component {
+  static defaultProps = {
+    initialValues: {},
+  };
+
+  state = {
+    loading: false,
+    user: this.props.user,
+  };
+
+  onConfirm = async () => {
+    const { user } = this.props;
+    this.setState({
+      loading: true,
+      error: null,
+    });
+    try {
+      const { data } = await request({
+        method: 'POST',
+        path: `/1/users/${user.id}/create-token`,
+      });
+      window.localStorage.setItem('tmpToken', data.token);
+      window.open(`/?switch-account=true`, '_blank');
```

Comment:
First off, the intent here is to temporarily only allow them impersonation in a single window (no tabs) correct? And that's what we want right? I'm kind of thinking that going forward we will want more permanent ongoing impersonation for things to work properly (ie what if you receive an email and click on a link?) but for now I'll concede this point.

Also it seems you can do this:

```js
const newWindow = window.open(...);
newWindow.sessionStorage.setItem(...);
```

would this not make more sense here?

Reply:
seems like it might apply below in the "little dance" section as well

Reply:
one last thing to keep in mind we often see localStorage being disabled in live sites (adblockers? I'm not 100% sure) ... I think maybe having it error out is OK here as for this operation it can't be recovered but just to keep in mind

Reply:
`First off, the intent here is to temporarily only allow them impersonation in a single window (no tabs) correct`
Correct

```
 And that's what we want right? I'm kind of thinking that going forward we will want more permanent ongoing impersonation for things to work properly (ie what if you receive an email and click on a link?) but for now I'll concede this point.
```

Fair enough, Its really meant as quick debug option, see what is going wrong of this given user. So far we have not experience any problems with this setup. That said we have not also asked for if this covers all the usecases. So i wont rule it out. Lets come back to that if we see there is a desire for more.

```
const newWindow = window.open(...);
newWindow.sessionStorage.setItem(...);
```

I will research, I was thinking that only worked for popups windows. 

```
one last thing to keep in mind we often see localStorage being disabled in live sites (adblockers? I'm not 100% sure) ... I think maybe having it error out is OK here as for this operation it can't be recovered but just to keep in mind
```

From what i can tell its not really a popup blocker things, more todo with mobile browser (specially the native app, web view) I have not gone 100% there yet. But from the errors i see they seem to be coming from mobile devices. 

That said I would like to switch to having in memory storage, to deal with that. But I would do that as separate thing. Its on my todo list.

I will kick off a card in linear to cover that. And we can start recording where those issues appears.
https://linear.app/bedrockdev/issue/BRC-34/deal-with-no-access-to-localstorage

Reply:
> Its really meant as quick debug option, see what is going wrong of this given user

Actually after thinking about it I think you're right this is probably a better solution. 👍

> more todo with mobile browser (specially the native app, web view) 

Ahh yes I think you're right! That's actually helpful I was wondering where all those errors were coming from.